sdk/python/feast/registry_server.py

```diff
@@ -32,23 +39,31 @@ def ApplyEntity(self, request: RegistryServer_pb2.ApplyEntityRequest, context):
         return Empty()
 
     def GetEntity(self, request: RegistryServer_pb2.GetEntityRequest, context):
-        return self.proxied_registry.get_entity(
+        entity = self.proxied_registry.get_entity(
             name=request.name, project=request.project, allow_cache=request.allow_cache
-        ).to_proto()
+        )
+        assert_permissions(resources=entity, actions=[AuthzedAction.READ])
+        return entity.to_proto()
 
     def ListEntities(self, request: RegistryServer_pb2.ListEntitiesRequest, context):
-        return RegistryServer_pb2.ListEntitiesResponse(
-            entities=[
-                entity.to_proto()
-                for entity in self.proxied_registry.list_entities(
-                    project=request.project,
-                    allow_cache=request.allow_cache,
-                    tags=dict(request.tags),
-                )
-            ]
+        entities = self.proxied_registry.list_entities(
+            project=request.project,
+            allow_cache=request.allow_cache,
+            tags=dict(request.tags),
         )
 
+        allowed_entities = []
```

Comment:
again, to simplify the list pattern, we can introduce a flag in the assert_permissions to filter out the unauthorized resources instead of returning a PermissionError as it is today. This can simplify and reduce the code to enforce permissions

Reply:
+1

Reply:
used `permitted_resources` function . Please take look again. Thanks